indico/modules/events/contributions/operations.py

```diff
@@ -153,3 +155,29 @@ def delete_subcontribution(subcontrib):
     logger.info('Subcontribution %s deleted by %s', subcontrib, session.user)
     subcontrib.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Subcontributions',
                              'Subcontribution "{}" has been deleted'.format(subcontrib.title), session.user)
+
+
+@no_autoflush
+def create_contribution_from_abstract(abstract, contrib_session=None):
+    event = abstract.event_new
+    contrib_person_links = set()
+    person_link_attrs = {'_title', 'address', 'affiliation', 'first_name', 'last_name', 'phone', 'author_type',
+                         'person', 'is_speaker'}
+    for abstract_person_link in abstract.person_links:
+        link = ContributionPersonLink()
+        link.populate_from_attrs(abstract_person_link, person_link_attrs)
+        contrib_person_links.add(link)
+
+    duration = contrib_session.default_contribution_duration if contrib_session else timedelta(minutes=15)
+    custom_fields_data = {'custom_{}'.format(field_value.contribution_field.id): field_value.data for
+                          field_value in abstract.field_values}
+    create_contribution(event, {'friendly_id': abstract.friendly_id,
```

Comment:
I would actually return the contribution here, not pass the `abstract` in the data dict and then assign it when calling this method:

    abstract.contribution = create_contribution_from_abstract(abstract, contrib_session)

That way the operations are less entangled with each other.